src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -795,17 +794,6 @@ private int decodeTransfer(
             assert sender != null; // TODO error if null
 
             server.decodableBodyBytes -= transfer.offset() - performative.offset();
-            if (!sender.fragmented)
-            {
-                assert deliveryId != NO_DELIVERY_ID; // TODO: error
-                assert deliveryId == sender.deliveryId + 1; // TODO: error
-                sender.deliveryId = deliveryId;
-            }
-
-            if (deliveryId != NO_DELIVERY_ID && deliveryId != sender.deliveryId) // TODO: error
-            {
-                break decode;
-            }
```

Comment:
Looks like we lost the check for invalid inbound `deliveryId`, this needs to be restored, right?
Please make sure to separate `remoteDeliveryId` coming from the client towards us from `deliveryId` generated by us, heading towards the client.